deps/coro-fs.lua

```diff
@@ -130,7 +131,22 @@ function fs.readFile(path)
   if err then return nil, err end
   stat, err = fs.fstat(fd)
   if stat then
-    data, err = fs.read(fd, stat.size)
+    --special case files on virtual filesystem
+    if stat.size == 0 and stat.birthtime.sec == 0 and stat.birthtime.nsec == 0 then
+      -- handle magic files the kernel generates as requested.
+      -- hopefully the heuristic works everywhere
+      local buffs = {}
+      local offs = 0
+      local size = 1024 * 48
+      repeat
+        data, err = fs.read(fd, size, offs)
```

Comment:
Won't this read the same chunk over and over? `offs` never changes.

Reply:
Yes, good point. I didn't test on a virtual file larger than 48 kilobytes. Fixing.

Reply:
Fixed